pkgs/development/python-modules/openusd/default.nix

```diff
@@ -0,0 +1,133 @@
+, ninja
+, setuptools
+, pyqt6
+, pyopengl
+, jinja2
+, pyside6
+, boost
+, numpy
+, git
+, tbb
+, opensubdiv
+, openimageio
+, opencolorio
+, osl
+, ptex
+, embree
+, openexr
+, flex
+, bison
+, qt6
+, python
+}:
+let
+  # Matches the pyside6-uic implementation
+  # https://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/pyside-tools/pyside_tool.py?id=e501cad66146a49c7a259579c7bb94bc93a67a08#n82
+  pyside-tools-uic = pkgs.writeShellScriptBin "pyside6-uic" ''
+    exec ${qt6.qtbase}/libexec/uic -g python "$@"
+  '';
+in
+buildPythonPackage rec {
+  pname = "OpenUSD";
+  version = "23.08";
+  src = fetchFromGitHub {
+    owner = "PixarAnimationStudios";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-3wM6stJnpznTU7Lb0vAdeO0/cr8n9rhvPZgT7PGAe04=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-for-pyside-6.5.0-initialization-order.patch";
+      url = "https://github.com/PixarAnimationStudios/OpenUSD/commit/78e66dee7dc448138687ce908ceeb2c4f54d83d9.patch";
+      hash = "sha256-0IRmqZABvrzJ4cza3c2f3sOaMVEwaoms3Up5RsjOXFs=";
+    })
+    (fetchpatch {
+      name = "fix-drawModeStandin-array.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/PixarAnimationStudios/OpenUSD/pull/2410.patch";
+      hash = "sha256-WDLUqK6/gKHv7kj2ipHfav/Z64uAXQGr8152vRHj3b8=";
+    })
+  ];
+
+  format = "other";
+
+  propagatedBuildInputs = [
+    setuptools
+    pyqt6
+    pyopengl
+    jinja2
+    pyside6
+    pyside-tools-uic
+    boost
+    numpy
+  ];
+
+  cmakeFlags = [
+    "-DPXR_BUILD_EXAMPLES=OFF"
+    "-DPXR_BUILD_TUTORIALS=OFF"
+    "-DPXR_BUILD_USD_TOOLS=ON"
+    "-DPXR_BUILD_IMAGING=ON"
+    "-DPXR_BUILD_USD_IMAGING=ON"
+    "-DPXR_BUILD_USDVIEW=ON"
+    "-DPXR_BUILD_DOCUMENTATION=ON"
+    "-DPXR_BUILD_PYTHON_DOCUMENTATION=ON"
+    "-DPXR_BUILD_EMBREE_PLUGIN=ON"
+    "-DPXR_BUILD_ALEMBIC_PLUGIN=ON"
+    # "-DPXR_ENABLE_MATERIALX_SUPPORT=ON"
+    "-DPXR_ENABLE_OSL_SUPPORT=ON"
+    "-DPXR_BUILD_DRACO_PLUGIN=ON"
+    "-DPXR_BUILD_MONOLITHIC=ON" # Seems to be commonly linked to monolithically
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    git
+    qt6.wrapQtAppsHook
+    doxygen
+    graphviz-nox
+  ];
+  buildInputs = [
+    tbb
+    opensubdiv
+    openimageio
+    opencolorio
+    osl
+    ptex
+    embree
+    pkgs.alembic.dev
+    openexr
+    flex
+    bison
+    boost
+    draco
+    qt6.qtbase
+    qt6.qtwayland
+  ];
+
+  pythonImportsCheck = [ "pxr" "pxr.Usd" ];
+
+  postInstall = ''
+    # Make python lib properly accessible
+    mkdir -p $out/${python.sitePackages}
+    cp -r $out/lib/python/* $out/${python.sitePackages}/
+  '';
+
+  meta = {
+    description = "Universal Scene Description";
+    longDescription = "Universal Scene Description (USD) is an efficient, scalable system for authoring, reading, and streaming time-sampled scene description for interchange between graphics applications.";
+    homepage = "https://openusd.org/";
+    license = lib.licenses.free;
```

Comment:
They seem to use an apache license, but they also bundle the licenses of their dependencies. I think it is safe to put `asl20` here